bindings/wasm/src/did/wasm_document.rs

```diff
@@ -87,10 +101,128 @@ impl WasmDocument {
     WasmDID(self.0.id().clone())
   }
 
+  /// Sets the controllers of the DID Document.
+  ///
+  /// Note: Duplicates will be ignored.
+  /// Use `null` to remove all controllers.
+  #[wasm_bindgen(js_name = setController)]
+  pub fn set_controller(&mut self, controllers: &UDID) -> Result<()> {
+    let controllers: Option<OneOrMany<IotaDID>> = controllers.into_serde().wasm_result()?;
+
+    let controller_set: Option<OneOrSet<IotaDID>> = if let Some(controllers) = controllers {
+      match controllers {
+        OneOrMany::One(controller) => Some(OneOrSet::new_one(controller)),
+        OneOrMany::Many(controllers) => {
+          if controllers.is_empty() {
+            None
+          } else {
+            let mut set: OrderedSet<IotaDID> = OrderedSet::new();
+            for controller in controllers {
+              set.append(controller);
+            }
+            Some(OneOrSet::new_set(set).wasm_result()?)
+          }
+        }
+      }
+    } else {
+      None
+    };
```

Comment:
This can be simplified.

```suggestion
    let controller_set: Option<OneOrSet<IotaDID>> = if let Some(controllers) = controllers.map(OneOrMany::into_vec) {
      if controllers.is_empty() {
        None
      } else {
        Some(OneOrSet::try_from(OrderedSet::from_iter(controllers)).wasm_result()?)
      }
    } else {
      None
    };
```

It can probably be simplified further with `map` but this is probably fine.